keyboards/redragon/k582/keyboard.json

```diff
@@ -0,0 +1,317 @@
+{
+    "keyboard_name": "K582 Surara",
+    "manufacturer": "Redragon",
+    "url": "https://redragonshop.com/products/surara-k582",
+    "maintainer": "Wallvon",
+    "processor": "SN32F248BF",
+    "bootloader": "sn32-dfu",
+    "usb": {
+        "vid": "0x0C45",
+        "pid": "0x5004",
+        "device_version": "1.0.0",
+        "force_nkro": true
+    },
+    "width": 21,
+    "height": 6,
+    "features": {
+        "audio": false,
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "debounce": 10,
+    "diode_direction": "ROW2COL",
+    "matrix_pins": {
+        "cols": ["A0","A1","A2","A3","A4","A5","A6","A7","A8","A9","A10","A11","A12","A13","A14","B0","B1","B2","B3","B4","B5"],
+        "rows": ["C15","D11","D10","D9","D8","D7"]
+    },
+    "indicators": {
+        "num_lock": "B13",
+        "caps_lock": "B14",
+        "scroll_lock": "B15",
+        "on_state": 0
+    },
+    "rgb_matrix": {
+        "animations": {
+            "breathing": true,
+            "band_spiral_val": true,
+            "cycle_all": true,
+            "cycle_left_right": true,
+            "cycle_up_down": true,
+            "rainbow_moving_chevron": true,
+            "cycle_out_in": true,
+            "cycle_out_in_dual": true,
+            "cycle_pinwheel": true,
+            "cycle_spiral": true,
+            "dual_beacon": true,
+            "rainbow_beacon": true,
+            "jellybean_raindrops": true,
+            "pixel_rain": true,
+            "typing_heatmap": true,
+            "solid_reactive_multiwide": true,
+            "solid_reactive_multinexus": true,
+            "splash": true,
+            "solid_splash": true,
+            "riverflow": true
+        },
+        "driver": "sn32f24xb",
+        "layout": [
+            {"matrix": [0, 0], "flags": 4, "x": 0, "y": 0},
```

Comment:
please update the flags according to each key function https://docs.qmk.fm/features/rgb_matrix#flags
alphas should be 4, mods 1, others as in the link